validation/citrus-validation-json/src/main/java/org/citrusframework/functions/core/JsonPatchFunction.java

```diff
@@ -0,0 +1,202 @@
+            pointer = pointer.substring(1);
+        }
+
+        Matcher matcher = ARRAY_INDEX_PATTERN.matcher(pointer);
+        StringBuilder sb = new StringBuilder();
+        while (matcher.find()) {
+            matcher.appendReplacement(sb, "/" + matcher.group(1));
+        }
+        matcher.appendTail(sb);
+        pointer = sb.toString();
+
+        pointer = pointer.replace(".", "/");
+
+        if (!pointer.startsWith("/")) {
+            pointer = "/" + pointer;
+        }
+
+        pointer = pointer.replace("//-", "/-");
+
+        return pointer;
+    }
+
+    private String escapeJsonString(String value) {
+        return value
+            .replace("\\", "\\\\")
+            .replace("\"", "\\\"")
+            .replace("\n", "\\n")
+            .replace("\r", "\\r")
+            .replace("\t", "\\t");
+    }
+
+    public record PatchOperation(String operation, String path, String value) {}
+
+    public static class Parameters implements FunctionParameters {
+        private String source;
+        private final List<PatchOperation> operations = new ArrayList<>();
+
+        @Override
+        public void configure(List<String> parameterList, TestContext context) {
+            if (parameterList == null || parameterList.isEmpty()) {
+                throw new InvalidFunctionUsageException("Function parameters must not be empty");
+            }
+
+            if (parameterList.size() < 4) {
+                throw new InvalidFunctionUsageException(
+                    "Missing parameters - usage: jsonPatch('jsonSource', 'operation', 'path', 'value', ...)");
+            }
+
+            if ((parameterList.size() - 1) % 3 != 0) {
+                throw new InvalidFunctionUsageException(
+                    "Invalid parameter count - operations must be provided as triplets (operation, path, value)");
+            }
+
+            setSource(parameterList.get(0));
+
+            for (int i = 1; i < parameterList.size(); i += 3) {
+                String operation = context.replaceDynamicContentInString(parameterList.get(i));
+                String path = context.replaceDynamicContentInString(parameterList.get(i + 1));
+                String value = context.replaceDynamicContentInString(parameterList.get(i + 2));
+
+                validateOperation(operation);
+                operations.add(new PatchOperation(operation, path, value));
+            }
+        }
+
+        private void validateOperation(String operation) {
+            if (!VALID_OPS.contains(operation)) {
+                throw new InvalidFunctionUsageException(
+                    "Invalid patch operation '" + operation + "'. Valid operations are: " + String.join(", ", VALID_OPS));
+            }
+        }
+
+        public String getSource() {
+            return source;
+        }
+
+        @SchemaProperty(required = true, description = "The JSON source content to be patched.")
+        public void setSource(String source) {
```

Comment:
Please add a setter for operations and annotate with SchemaProperty information. Would be good to add SchemaProperty annotations also for the PatchOperation setters.

This is quite new to generate a schema for the Citrus function parameters. The generated schema is used in YAML DSL and the Kaoto graphical UI designer.